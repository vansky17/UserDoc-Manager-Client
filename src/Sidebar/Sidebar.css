nav{
  overflow-y: auto;  
  z-index: 2;
}
@media screen and (max-width:820px){
  nav{
    overflow-y: auto;  
  }
}
@media screen and (min-width:670px){
  .divider-menu{
    display:none;
  }
}
