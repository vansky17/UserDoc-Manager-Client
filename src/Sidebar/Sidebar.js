import React from "react";
/* import { Route, Link } from "react-router-dom"; */
import UserDocsContext from "../UserDocsContext";
import CircleButton from '../CircleButton/CircleButton'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faHome, faFileUpload, faProjectDiagram } from "@fortawesome/free-solid-svg-icons";
import { HashLink as Link  } from 'react-router-hash-link';
import "./App.css";
import Config from '../config'
const API = Config.API_ENDPOINT;

class Sidebar extends React.Component {
  static contextType = UserDocsContext;
  render () {
    const { products=[] } = this.context
    return(
      <>
        <nav role="navigation">
        <div className="hamburger-menu">
      <input type="checkbox" id="hamburger"/>
      <label htmlFor="hamburger" className="hamburger">
        <span className="line"></span>
        <span className="line"></span>
        <span className="line"></span>
      </label>
    </div>
    <div className="menu">
      <div>
        <ul>
          <li>
            <CircleButton   
                tag={Link}
                to='/#home'
                type='button'
                className='NoteListNav__add-folder-button'
              >
                <FontAwesomeIcon icon={faHome} />
                <br />
                  HOME
              </CircleButton>
            </li>
            {products.map(product =>
            <li key={product.id}>
            <CircleButton
              tag={Link}
              className='NoteListNav__folder-link'
              to={`/#${product.id}`}
              type='button'
            >
              {product.name}
            </CircleButton>
            </li>
          )}
          <hr></hr>
          <li>
        <CircleButton   
            tag={Link}
            to='/upload-file'
            type='button'
            /* className='NoteListNav__add-folder-button' */
          >
            <FontAwesomeIcon icon={faFileUpload} />
            <br />
            UPLOAD DOC
          </CircleButton>
        <CircleButton   
            tag={Link}
            to='/add-product'
            type='button'
            className=''
          >
            <FontAwesomeIcon icon={faProjectDiagram} />
            <br />
            MANAGE PRODUCTS
          </CircleButton>
        </li>
        </ul>
      </div>
    </div>
    <div className="divider-menu"></div>
    <div className="mobile-menu slideDown">
      <ul>
        {/* For the static version: 
        <li>
          <a href="#home">Home</a>
        </li>
        <li>
          <a href="#p1">Product 1</a>
        </li>
        <li>
          <a href="#p2">Product 2</a>
        </li>
        <li>
          <a href="#p3">Product 3</a>
        </li>
        <hr></hr>
        <li>
        <CircleButton   
            tag={Link}
            to='/upload-file'
            type='button'
            className='NoteListNav__add-folder-button'
          >
            <FontAwesomeIcon icon='plus' />
            <br />
            UPLOAD
          </CircleButton>
        </li>
        <li>
          <a href="productgroups.html">Add Product Group</a>
        </li> */}
      </ul>
    </div>
        </nav>
      </>
    )
  }
}
export default Sidebar