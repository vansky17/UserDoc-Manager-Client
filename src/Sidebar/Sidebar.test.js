import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Sidebar from './Sidebar'

describe(`Doc component`, () => {
  const props = {
  docs: [
    {
      "id": 24,
      "name": "practice for mockup2",
      "partnum": "34563456",
      "vernum": 1,
      "formattype": "mp4",
      "reldate": "2020-02-12T21:27:48.510Z",
      "author": "John Doe",
      "productid": 6,
      "descr": "No description",
      "path": "https://userdocsmanager.s3.us-east-2.amazonaws.com/notes.png"
  },
  {
      "id": 25,
      "name": "Video",
      "partnum": "127",
      "vernum": 1,
      "formattype": "mp4",
      "reldate": "2020-02-12T21:31:02.972Z",
      "author": "IS",
      "productid": 6,
      "descr": "No description",
      "path": "https://userdocsmanager.s3.us-east-2.amazonaws.com/small.mp4"
  },
  {
      "id": 26,
      "name": "JS-playground 2",
      "partnum": "123.4567",
      "vernum": 1,
      "formattype": "DOCX",
      "reldate": "2020-02-12T22:28:23.585Z",
      "author": "IS",
      "productid": 6,
      "descr": "No description",
      "path": "https://userdocsmanager.s3.us-east-2.amazonaws.com/clean_form_bg.jpg"
  }
  ],
  products: [
    {
      "id": 1,
      "name": "Product Line 1"
    },
    {
      "id": 2,
      "name": "Product Line 2"
    },
    {
      "id": 3,
      "name": "Third Product Line"
    },
    {
      "id": 6,
      "name": "Test Ground"
    }
  ]
}

  it('renders a .Sidebar by default', () => {
    const wrapper = shallow(<Sidebar />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the Sidebar given props', () => {
    const wrapper = shallow(<Sidebar {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})