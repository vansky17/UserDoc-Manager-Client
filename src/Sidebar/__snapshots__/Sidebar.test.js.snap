// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Doc component renders a .Sidebar by default 1`] = `
<nav
  role="navigation"
>
  <div
    className="hamburger-menu"
  >
    <input
      id="hamburger"
      onClick={[Function]}
      type="checkbox"
    />
    <label
      className="hamburger"
      htmlFor="hamburger"
    >
      <span
        className="line"
      />
      <span
        className="line"
      />
      <span
        className="line"
      />
    </label>
  </div>
  <div
    className="menu"
  >
    <div>
      <ul>
        <li>
          <NavCircleButton
            className=""
            tag={[Function]}
            to="/#home"
            type="button"
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    576,
                    512,
                    Array [],
                    "f015",
                    "M488 312.7V456c0 13.3-10.7 24-24 24H348c-6.6 0-12-5.4-12-12V356c0-6.6-5.4-12-12-12h-72c-6.6 0-12 5.4-12 12v112c0 6.6-5.4 12-12 12H112c-13.3 0-24-10.7-24-24V312.7c0-3.6 1.6-7 4.4-9.3l188-154.8c4.4-3.6 10.8-3.6 15.3 0l188 154.8c2.7 2.3 4.3 5.7 4.3 9.3zm83.6-60.9L488 182.9V44.4c0-6.6-5.4-12-12-12h-56c-6.6 0-12 5.4-12 12V117l-89.5-73.7c-17.7-14.6-43.3-14.6-61 0L4.4 251.8c-5.1 4.2-5.8 11.8-1.6 16.9l25.5 31c4.2 5.1 11.8 5.8 16.9 1.6l235.2-193.7c4.4-3.6 10.8-3.6 15.3 0l235.2 193.7c5.1 4.2 12.7 3.5 16.9-1.6l25.5-31c4.2-5.2 3.4-12.7-1.7-16.9z",
                  ],
                  "iconName": "home",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
            <br />
            HOME
          </NavCircleButton>
        </li>
        <hr />
        <li>
          <NavCircleButton
            className=""
            tag={[Function]}
            to="/upload-file#add-doc"
            type="button"
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    384,
                    512,
                    Array [],
                    "f574",
                    "M224 136V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V160H248c-13.2 0-24-10.8-24-24zm65.18 216.01H224v80c0 8.84-7.16 16-16 16h-32c-8.84 0-16-7.16-16-16v-80H94.82c-14.28 0-21.41-17.29-11.27-27.36l96.42-95.7c6.65-6.61 17.39-6.61 24.04 0l96.42 95.7c10.15 10.07 3.03 27.36-11.25 27.36zM377 105L279.1 7c-4.5-4.5-10.6-7-17-7H256v128h128v-6.1c0-6.3-2.5-12.4-7-16.9z",
                  ],
                  "iconName": "file-upload",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
            <br />
            UPLOAD DOC
          </NavCircleButton>
          <NavCircleButton
            className=""
            tag={[Function]}
            to="/add-product#add-pg"
            type="button"
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    640,
                    512,
                    Array [],
                    "f542",
                    "M384 320H256c-17.67 0-32 14.33-32 32v128c0 17.67 14.33 32 32 32h128c17.67 0 32-14.33 32-32V352c0-17.67-14.33-32-32-32zM192 32c0-17.67-14.33-32-32-32H32C14.33 0 0 14.33 0 32v128c0 17.67 14.33 32 32 32h95.72l73.16 128.04C211.98 300.98 232.4 288 256 288h.28L192 175.51V128h224V64H192V32zM608 0H480c-17.67 0-32 14.33-32 32v128c0 17.67 14.33 32 32 32h128c17.67 0 32-14.33 32-32V32c0-17.67-14.33-32-32-32z",
                  ],
                  "iconName": "project-diagram",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
            <br />
            MANAGE PRODUCTS
          </NavCircleButton>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="divider-menu"
  >
    <div
      className="mobile-menu slideDown"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <ul>
        <li>
          <NavCircleButton
            className=""
            onClick={[Function]}
            tag={[Function]}
            to="/#home"
            type="button"
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    576,
                    512,
                    Array [],
                    "f015",
                    "M488 312.7V456c0 13.3-10.7 24-24 24H348c-6.6 0-12-5.4-12-12V356c0-6.6-5.4-12-12-12h-72c-6.6 0-12 5.4-12 12v112c0 6.6-5.4 12-12 12H112c-13.3 0-24-10.7-24-24V312.7c0-3.6 1.6-7 4.4-9.3l188-154.8c4.4-3.6 10.8-3.6 15.3 0l188 154.8c2.7 2.3 4.3 5.7 4.3 9.3zm83.6-60.9L488 182.9V44.4c0-6.6-5.4-12-12-12h-56c-6.6 0-12 5.4-12 12V117l-89.5-73.7c-17.7-14.6-43.3-14.6-61 0L4.4 251.8c-5.1 4.2-5.8 11.8-1.6 16.9l25.5 31c4.2 5.1 11.8 5.8 16.9 1.6l235.2-193.7c4.4-3.6 10.8-3.6 15.3 0l235.2 193.7c5.1 4.2 12.7 3.5 16.9-1.6l25.5-31c4.2-5.2 3.4-12.7-1.7-16.9z",
                  ],
                  "iconName": "home",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
            <br />
            HOME
          </NavCircleButton>
        </li>
        <li
          style={
            Object {
              "alignContent": "center ",
              "display": "flex",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "space-between",
            }
          }
        >
           
        </li>
        <hr />
        <li>
          <NavCircleButton
            className=""
            onClick={[Function]}
            tag={[Function]}
            to="/upload-file#add-doc"
            type="button"
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    384,
                    512,
                    Array [],
                    "f574",
                    "M224 136V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V160H248c-13.2 0-24-10.8-24-24zm65.18 216.01H224v80c0 8.84-7.16 16-16 16h-32c-8.84 0-16-7.16-16-16v-80H94.82c-14.28 0-21.41-17.29-11.27-27.36l96.42-95.7c6.65-6.61 17.39-6.61 24.04 0l96.42 95.7c10.15 10.07 3.03 27.36-11.25 27.36zM377 105L279.1 7c-4.5-4.5-10.6-7-17-7H256v128h128v-6.1c0-6.3-2.5-12.4-7-16.9z",
                  ],
                  "iconName": "file-upload",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
            <br />
            UPLOAD DOC
          </NavCircleButton>
          <NavCircleButton
            className=""
            onClick={[Function]}
            tag={[Function]}
            to="/add-product#add-pg"
            type="button"
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    640,
                    512,
                    Array [],
                    "f542",
                    "M384 320H256c-17.67 0-32 14.33-32 32v128c0 17.67 14.33 32 32 32h128c17.67 0 32-14.33 32-32V352c0-17.67-14.33-32-32-32zM192 32c0-17.67-14.33-32-32-32H32C14.33 0 0 14.33 0 32v128c0 17.67 14.33 32 32 32h95.72l73.16 128.04C211.98 300.98 232.4 288 256 288h.28L192 175.51V128h224V64H192V32zM608 0H480c-17.67 0-32 14.33-32 32v128c0 17.67 14.33 32 32 32h128c17.67 0 32-14.33 32-32V32c0-17.67-14.33-32-32-32z",
                  ],
                  "iconName": "project-diagram",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
            <br />
            MANAGE PRODUCTS
          </NavCircleButton>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;

exports[`Doc component renders the Sidebar given props 1`] = `
<nav
  role="navigation"
>
  <div
    className="hamburger-menu"
  >
    <input
      id="hamburger"
      onClick={[Function]}
      type="checkbox"
    />
    <label
      className="hamburger"
      htmlFor="hamburger"
    >
      <span
        className="line"
      />
      <span
        className="line"
      />
      <span
        className="line"
      />
    </label>
  </div>
  <div
    className="menu"
  >
    <div>
      <ul>
        <li>
          <NavCircleButton
            className=""
            tag={[Function]}
            to="/#home"
            type="button"
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    576,
                    512,
                    Array [],
                    "f015",
                    "M488 312.7V456c0 13.3-10.7 24-24 24H348c-6.6 0-12-5.4-12-12V356c0-6.6-5.4-12-12-12h-72c-6.6 0-12 5.4-12 12v112c0 6.6-5.4 12-12 12H112c-13.3 0-24-10.7-24-24V312.7c0-3.6 1.6-7 4.4-9.3l188-154.8c4.4-3.6 10.8-3.6 15.3 0l188 154.8c2.7 2.3 4.3 5.7 4.3 9.3zm83.6-60.9L488 182.9V44.4c0-6.6-5.4-12-12-12h-56c-6.6 0-12 5.4-12 12V117l-89.5-73.7c-17.7-14.6-43.3-14.6-61 0L4.4 251.8c-5.1 4.2-5.8 11.8-1.6 16.9l25.5 31c4.2 5.1 11.8 5.8 16.9 1.6l235.2-193.7c4.4-3.6 10.8-3.6 15.3 0l235.2 193.7c5.1 4.2 12.7 3.5 16.9-1.6l25.5-31c4.2-5.2 3.4-12.7-1.7-16.9z",
                  ],
                  "iconName": "home",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
            <br />
            HOME
          </NavCircleButton>
        </li>
        <hr />
        <li>
          <NavCircleButton
            className=""
            tag={[Function]}
            to="/upload-file#add-doc"
            type="button"
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    384,
                    512,
                    Array [],
                    "f574",
                    "M224 136V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V160H248c-13.2 0-24-10.8-24-24zm65.18 216.01H224v80c0 8.84-7.16 16-16 16h-32c-8.84 0-16-7.16-16-16v-80H94.82c-14.28 0-21.41-17.29-11.27-27.36l96.42-95.7c6.65-6.61 17.39-6.61 24.04 0l96.42 95.7c10.15 10.07 3.03 27.36-11.25 27.36zM377 105L279.1 7c-4.5-4.5-10.6-7-17-7H256v128h128v-6.1c0-6.3-2.5-12.4-7-16.9z",
                  ],
                  "iconName": "file-upload",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
            <br />
            UPLOAD DOC
          </NavCircleButton>
          <NavCircleButton
            className=""
            tag={[Function]}
            to="/add-product#add-pg"
            type="button"
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    640,
                    512,
                    Array [],
                    "f542",
                    "M384 320H256c-17.67 0-32 14.33-32 32v128c0 17.67 14.33 32 32 32h128c17.67 0 32-14.33 32-32V352c0-17.67-14.33-32-32-32zM192 32c0-17.67-14.33-32-32-32H32C14.33 0 0 14.33 0 32v128c0 17.67 14.33 32 32 32h95.72l73.16 128.04C211.98 300.98 232.4 288 256 288h.28L192 175.51V128h224V64H192V32zM608 0H480c-17.67 0-32 14.33-32 32v128c0 17.67 14.33 32 32 32h128c17.67 0 32-14.33 32-32V32c0-17.67-14.33-32-32-32z",
                  ],
                  "iconName": "project-diagram",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
            <br />
            MANAGE PRODUCTS
          </NavCircleButton>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="divider-menu"
  >
    <div
      className="mobile-menu slideDown"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <ul>
        <li>
          <NavCircleButton
            className=""
            onClick={[Function]}
            tag={[Function]}
            to="/#home"
            type="button"
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    576,
                    512,
                    Array [],
                    "f015",
                    "M488 312.7V456c0 13.3-10.7 24-24 24H348c-6.6 0-12-5.4-12-12V356c0-6.6-5.4-12-12-12h-72c-6.6 0-12 5.4-12 12v112c0 6.6-5.4 12-12 12H112c-13.3 0-24-10.7-24-24V312.7c0-3.6 1.6-7 4.4-9.3l188-154.8c4.4-3.6 10.8-3.6 15.3 0l188 154.8c2.7 2.3 4.3 5.7 4.3 9.3zm83.6-60.9L488 182.9V44.4c0-6.6-5.4-12-12-12h-56c-6.6 0-12 5.4-12 12V117l-89.5-73.7c-17.7-14.6-43.3-14.6-61 0L4.4 251.8c-5.1 4.2-5.8 11.8-1.6 16.9l25.5 31c4.2 5.1 11.8 5.8 16.9 1.6l235.2-193.7c4.4-3.6 10.8-3.6 15.3 0l235.2 193.7c5.1 4.2 12.7 3.5 16.9-1.6l25.5-31c4.2-5.2 3.4-12.7-1.7-16.9z",
                  ],
                  "iconName": "home",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
            <br />
            HOME
          </NavCircleButton>
        </li>
        <li
          style={
            Object {
              "alignContent": "center ",
              "display": "flex",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "space-between",
            }
          }
        >
           
        </li>
        <hr />
        <li>
          <NavCircleButton
            className=""
            onClick={[Function]}
            tag={[Function]}
            to="/upload-file#add-doc"
            type="button"
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    384,
                    512,
                    Array [],
                    "f574",
                    "M224 136V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V160H248c-13.2 0-24-10.8-24-24zm65.18 216.01H224v80c0 8.84-7.16 16-16 16h-32c-8.84 0-16-7.16-16-16v-80H94.82c-14.28 0-21.41-17.29-11.27-27.36l96.42-95.7c6.65-6.61 17.39-6.61 24.04 0l96.42 95.7c10.15 10.07 3.03 27.36-11.25 27.36zM377 105L279.1 7c-4.5-4.5-10.6-7-17-7H256v128h128v-6.1c0-6.3-2.5-12.4-7-16.9z",
                  ],
                  "iconName": "file-upload",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
            <br />
            UPLOAD DOC
          </NavCircleButton>
          <NavCircleButton
            className=""
            onClick={[Function]}
            tag={[Function]}
            to="/add-product#add-pg"
            type="button"
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    640,
                    512,
                    Array [],
                    "f542",
                    "M384 320H256c-17.67 0-32 14.33-32 32v128c0 17.67 14.33 32 32 32h128c17.67 0 32-14.33 32-32V352c0-17.67-14.33-32-32-32zM192 32c0-17.67-14.33-32-32-32H32C14.33 0 0 14.33 0 32v128c0 17.67 14.33 32 32 32h95.72l73.16 128.04C211.98 300.98 232.4 288 256 288h.28L192 175.51V128h224V64H192V32zM608 0H480c-17.67 0-32 14.33-32 32v128c0 17.67 14.33 32 32 32h128c17.67 0 32-14.33 32-32V32c0-17.67-14.33-32-32-32z",
                  ],
                  "iconName": "project-diagram",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
            <br />
            MANAGE PRODUCTS
          </NavCircleButton>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
